src/main/controllers/BulkCreateMediaAccountsConfirmationController.ts

```diff
@@ -0,0 +1,79 @@
+import {PipRequest} from '../models/request/PipRequest';
+import {Response} from 'express';
+import {cloneDeep} from 'lodash';
+import {FileHandlingService} from '../service/fileHandlingService';
+import {CreateAccountService} from '../service/createAccountService';
+
+const createAccountService = new CreateAccountService();
+const fileHandlingService = new FileHandlingService();
+
+const bulkCreateAccountsUrl = 'bulk-create-media-accounts';
+const bulkCreateAccountsConfirmationUrl = 'bulk-create-media-accounts-confirmation';
+const bulkCreateAccountsConfirmedUrl = 'bulk-create-media-accounts-confirmed';
+
+export default class BulkCreateMediaAccountsConfirmationController {
+  public async get(req: PipRequest, res: Response): Promise<void> {
+    const formData = (req.cookies?.formCookie) ? JSON.parse(req.cookies['formCookie']) : {};
+    const fileName = formData['uploadFileName'];
+
+    const accountsToCreate =
+      fileName === undefined ? []
+        : await BulkCreateMediaAccountsConfirmationController.getAccountsToCreate(req.user['oid'], fileName);
+
+    res.render(bulkCreateAccountsConfirmationUrl, {
+      ...cloneDeep(req.i18n.getDataByLanguage(req.lng)[bulkCreateAccountsConfirmationUrl]),
+      accountsToCreate,
+    });
+  }
+
+  public async post(req: PipRequest, res: Response): Promise<void> {
+    const confirmed = req.body['confirmed'];
+    const formData = (req.cookies?.formCookie) ? JSON.parse(req.cookies['formCookie']) : {};
+    const fileName = formData.uploadFileName;
+    const accountsToCreate = await BulkCreateMediaAccountsConfirmationController.getAccountsToCreate(req.user['oid'], fileName);
+
+    if (!confirmed) {
+      return res.render(bulkCreateAccountsConfirmationUrl, {
+        ...cloneDeep(req.i18n.getDataByLanguage(req.lng)[bulkCreateAccountsConfirmationUrl]),
+        accountsToCreate,
+        displayNoOptionError: true,
+      });
+    }
+
+    if (confirmed === 'Yes') {
+      const file = await fileHandlingService.readFileFromRedis(req.user['oid'], fileName);
+      const success = await createAccountService.bulkCreateMediaAccounts(file, fileName, req.user?.['piUserId']);
+      if (success) {
+        return res.redirect(bulkCreateAccountsConfirmedUrl);
+      } else {
+        return res.render(bulkCreateAccountsConfirmationUrl, {
+          ...cloneDeep(req.i18n.getDataByLanguage(req.lng)[bulkCreateAccountsConfirmationUrl]),
+          accountsToCreate,
+          displayAccountCreationError: true,
+        });
+      }
+    } else {
+      return res.redirect(bulkCreateAccountsUrl);
+    }
+  }
+
+  private static async getAccountsToCreate(userId, fileName): Promise<string[][]> {
+    const file = await fileHandlingService.readFileFromRedis(userId, fileName);
+    const fileData = fileHandlingService.readCsvToArray(file);
+    return BulkCreateMediaAccountsConfirmationController.reorderFieldsInRows(fileData);
+  }
+
+  private static reorderFieldsInRows(rows): string[][] {
```

Comment:
The backend doesn't need a specific order, so wondering if we can remove this below?

Reply:
There is a table on that confirmation screen showing the list of media accounts to be created. We need the ordering here so the table header matches the table contents :)